.github/workflows/nextjs_bundle_analysis.yml

```diff
@@ -0,0 +1,117 @@
+name: 'Next.js Bundle Analysis'
+
+on:
+   pull_request:
+   push:
+      branches:
+         - main # change this if your default branch is named differently
+   workflow_dispatch:
+
+defaults:
+   run:
+      # change this if your nextjs app does not live at the root of the repo
+      working-directory: ./frontend
+
+jobs:
+   analyze:
+      runs-on: ubuntu-latest
+      steps:
+         - name: Checkout
+           uses: actions/checkout@v2
+
+         - name: Setup and cache pnpm
+           uses: ./.github/actions/pnpm-setup
+
+         - name: pnpm install
+           run: pnpm install
+           env:
+              FONT_AWESOME_NPM_TOKEN: ${{ secrets.FONT_AWESOME_NPM_TOKEN }}
+
+         - name: Restore next build
+           uses: actions/cache@v2
+           id: restore-build-cache
+           env:
+              cache-name: cache-next-build
+           with:
+              # if you use a custom build directory, replace all instances of `.next` in this file with your build directory
+              # ex: if your app builds to `dist`, replace `.next` with `dist`
+              path: .next/cache
+              # change this if you prefer a more strict cache
+              key: ${{ runner.os }}-build-${{ env.cache-name }}
+
+         - name: Build next.js app
+           # change this if your site requires a custom build command
+           run: ./node_modules/.bin/next build
+           env:
+              ALGOLIA_API_KEY: ${{ secrets.ALGOLIA_API_KEY }}
+              NEXT_PUBLIC_STRAPI_ORIGIN: https://main.govinor.com
+
+         # Here's the first place where next-bundle-analysis' own script is used
+         # This step pulls the raw bundle stats for the current bundle
+         - name: Analyze bundle
+           run: npx -p nextjs-bundle-analysis report
+
+         - name: Upload bundle
+           uses: actions/upload-artifact@v2
+           with:
+              name: bundle
+              path: .next/analyze/__bundle_analysis.json
+
+         - name: Download base branch bundle stats
+           uses: dawidd6/action-download-artifact@v2
```

Comment:
TIL! @mlahargou apparently this GH Action allows you to download artifacts from other workflows.  https://github.com/dawidd6/action-download-artifact

> Let's suppose you have a workflow with a job in it that at the end uploads an artifact using actions/upload-artifact action and you want to download this artifact in another workflow that is run after the first one. Official actions/download-artifact does not allow this. That's why I decided to create this action.

Reply:
Cool!